h2o-core/src/main/java/hex/Model.java

```diff
@@ -2008,21 +2008,19 @@ public void map(Chunk[] cs, NewChunk[] nc) {
     String [] names = new String[ncols];
     if(output.hasTreatment()){
       names[0] = "uplift_predict";
+      names[1] = "p_y1_ct1";
+      names[2] = "p_y1_ct0";
     } else {
       names[0] = "predict";
-    }
-    for(int i = 1; i < names.length; ++i) {
-      names[i] = output.classNames()[i - 1];
-      // turn integer class labels such as 0, 1, etc. into p0, p1, etc.
-      try {
-        Integer.valueOf(names[i]);
-        if(output.hasTreatment()){
-          names[i] = i == 1? "p_y1_ct1" : "p_y1_ct0";
-        } else {
+      for (int i = 1; i < names.length; ++i) {
+        names[i] = output.classNames()[i - 1];
+        // turn integer class labels such as 0, 1, etc. into p0, p1, etc.
+        try {
+          Integer.valueOf(names[i]);
           names[i] = "p" + names[i];
+        } catch (Throwable t) {
```

Comment:
the try-catch should only wrap Integer.valueOf

Reply:
or is this just a revert of how it was before?

Reply:
Yes, now the uplift case is separate and the other code is the same as before my uplift changes.

Reply:
ok, lets leave it as-is then - thanks for explaining!